src/nvim/charset.c

```diff
@@ -461,8 +451,8 @@ char_u* str_foldcase(char_u *str, int orglen, char_u *buf, int buflen)
 
             if (olen != nlen) {
               if (buf == NULL) {
-                STRMOVE(GA_PTR(i) + nlen, GA_PTR(i) + olen);
-                ga.ga_len += nlen - olen;
+                STRMOVE(ks.s + i + nlen, ks.s + i + olen);
+                ks.l += nlen - olen;
```

Comment:
Garray is not limited only by strings. It is more like kvec, except that you have more control over how garray grows.